### PR TITLE
Adding keycloak

### LINKED ISSUE #1864: [Add] Adding link of Keycloak
### Site URL

https://www.keycloak.org/

### Description of the above provided site

Add authentication to applications and secure services with minimum effort. There is no need to deal with storing or authenticating users. Keycloak provides user federation, strong authentication, user management, fine-grained authorization, and more.

### Category

Backend

### Sub Category

Authentication

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by Prakhar-Shankar**
## Fixes Issue

Closes #1864 

## Changes proposed

Keycloak is a authentication service provider.
Adding this in category - `Backend`,  sub-category- `Authentication`

### CODE REVIEW

**Comment by CBID2** [comment]
Hi @Prakhar-Shankar. Welcome to LinksHub! :) Even though you've raised an issue, it has not been approved or assigned to you yet. For now, your PR will be in draft mode, and once it's approved, you'll get a review.  In the future, please wait until your issue has been approved and assigned to you before you submit a pull request. Happy Contributing! 😊

File: database/backend/authentication.json [hunk 1]
```diff
@@ -47,5 +47,12 @@
     "url": "https://kinde.com/",
     "category": "backend",
     "subcategory": "authentication"
+  },
+  {
+    "name": "Keycloak",
+    "description": "Add authentication to applications and secure services with minimum effort. There is no need to deal with storing or authenticating users. Keycloak provides user federation, strong authentication, user management, fine-grained authorization, and more.",
```
Comment (CBID2) [inline database/backend/authentication.json:53]: [proofreading/suggestion] So far, this description describes the function of Keycloak. However, the first two sentences describing authentication in general terms make that point unclear.  
```suggestion
    "description": "Keycloak provides user federation, strong authentication, user management, fine-grained authorization, and more.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @Prakhar-Shankar! :) I pointed out a suggestion in my inline comment. In the future, consider focusing more on the resource's purpose when describing them (and running them through a tool like Grammarly for grammar errors) before making a pull request. Other than that, you're off to a great start! :)

**Review by CBID2 — APPROVED** [review]
Well done @Prakhar-Shankar! :) @Anmol-Baranwal, it's your turn.

**Comment by Prakhar-Shankar** [comment]
> Well done @Prakhar-Shankar! :) @Anmol-Baranwal, it's your turn.

Thank you for the encouragement.

**Comment by CBID2** [comment]
> > Well done @Prakhar-Shankar! :) @Anmol-Baranwal, it's your turn.
> 
> Thank you for the encouragement.

Anytime my friend! :)

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks good to me.

**Pull request merged**